https://bugs.gentoo.org/723468#c4
By Anon Emuss <abuslbea@centurylink.net>
----
Update tests for sphinx-3

The deprecated source_parsers no longer works with sphinx-3.  Modify
conf.py files to use the recommonmark extension, which sphinx supports.

Tested and appears to work with sphinx-2.4.4 and sphinx-3.2.1.

diff --git a/tests/sphinx_code_block/conf.py b/tests/sphinx_code_block/conf.py
index 6b0140e..ddfafd5 100644
--- a/tests/sphinx_code_block/conf.py
+++ b/tests/sphinx_code_block/conf.py
@@ -1,11 +1,9 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
-
 templates_path = ['_templates']
-source_suffix = '.md'
-source_parsers = { '.md': CommonMarkParser }
+extensions = ['recommonmark']
+source_suffix = { '.md': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
diff --git a/tests/sphinx_custom_md/conf.py b/tests/sphinx_custom_md/conf.py
index b29157b..9073346 100644
--- a/tests/sphinx_custom_md/conf.py
+++ b/tests/sphinx_custom_md/conf.py
@@ -1,12 +1,11 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
 from recommonmark.transform import AutoStructify
 
 templates_path = ['_templates']
-source_suffix = '.markdown'
-source_parsers = { '.markdown': CommonMarkParser }
+extensions = ['recommonmark']
+source_suffix = { '.markdown': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
diff --git a/tests/sphinx_generic/conf.py b/tests/sphinx_generic/conf.py
index 6b0140e..ddfafd5 100644
--- a/tests/sphinx_generic/conf.py
+++ b/tests/sphinx_generic/conf.py
@@ -1,11 +1,9 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
-
 templates_path = ['_templates']
-source_suffix = '.md'
-source_parsers = { '.md': CommonMarkParser }
+extensions = ['recommonmark']
+source_suffix = { '.md': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
diff --git a/tests/sphinx_indented_code/conf.py b/tests/sphinx_indented_code/conf.py
index f441071..90c5692 100644
--- a/tests/sphinx_indented_code/conf.py
+++ b/tests/sphinx_indented_code/conf.py
@@ -1,11 +1,9 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
-
 templates_path = ['_templates']
-source_suffix = '.md'
-source_parsers = { '.md': CommonMarkParser }
+extensions = ['recommonmark']
+source_suffix = { '.md': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
diff --git a/tests/sphinx_nested_header_block/conf.py b/tests/sphinx_nested_header_block/conf.py
index 6b0140e..ddfafd5 100644
--- a/tests/sphinx_nested_header_block/conf.py
+++ b/tests/sphinx_nested_header_block/conf.py
@@ -1,11 +1,9 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
-
 templates_path = ['_templates']
-source_suffix = '.md'
-source_parsers = { '.md': CommonMarkParser }
+extensions = ['recommonmark']
+source_suffix = { '.md': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
diff --git a/tests/sphinx_xref/conf.py b/tests/sphinx_xref/conf.py
index 443a026..acd840b 100644
--- a/tests/sphinx_xref/conf.py
+++ b/tests/sphinx_xref/conf.py
@@ -1,14 +1,11 @@
 
 # -*- coding: utf-8 -*-
 
-from recommonmark.parser import CommonMarkParser
-
-extensions = 'sphinx.ext.autosectionlabel']
+extensions = ['sphinx.ext.autosectionlabel', 'recommonmark']
 autosectionlabel_prefix_document = True
 
 templates_path = ['_templates']
-source_suffix = '.md'
-source_parsers = { '.md': CommonMarkParser }
+source_suffix = { '.md': 'markdown' }
 master_doc = 'index'
 project = u'sphinxproj'
 copyright = u'2015, rtfd'
